sdk/go/common/workspace/plugins_test.go

```diff
@@ -1429,6 +1429,59 @@ func TestPluginSpec_GetSource(t *testing.T) {
 	}
 }
 
+//nolint:paralleltest // mutates pluginDownloadURLOverridesParsed
+func TestFallbackSource_URLOverride(t *testing.T) {
+	// Test that fallbackSource respects URL overrides for get.pulumi.com
+	// when falling back from GitHub.
+
+	// Create a test server to serve as the override destination
+	var requestReceived bool
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		requestReceived = true
+		t.Logf("Test server received request: %s", r.URL.String())
+		// Return a 404 to simulate plugin not found
+		w.WriteHeader(http.StatusNotFound)
+	}))
+	t.Cleanup(server.Close)
+
+	// Set up URL override for get.pulumi.com to point to our test server
+	pluginDownloadURLOverridesParsed = pluginDownloadOverrideArray{
+		{
+			reg: regexp.MustCompile(`^https://get\.pulumi\.com/releases/plugins$`),
+			url: server.URL,
+		},
+	}
+	t.Cleanup(func() {
+		pluginDownloadURLOverridesParsed = nil
+	})
+
+	// Create a fallbackSource
+	source := newFallbackSource("test-plugin", apitype.ResourcePlugin)
+
+	// Attempt to download - this should fail at GitHub, then try get.pulumi.com,
+	// which should be overridden to our test server
+	version := semver.MustParse("1.0.0")
+	_, _, err := source.Download(context.Background(), version, "linux", "amd64", func(req *http.Request) (io.ReadCloser, int64, error) {
+		t.Logf("Download attempt to: %s", req.URL.String())
+		resp, err := http.DefaultClient.Do(req)
```

Comment:
Hrm we don't actually need to do the request here if the URL is a GitHub url right?  That would avoid the round trip, but more importantly also any future clashes if we decide to create a plugin named `test-plugin`.

We can just parse the URL here, and if it contains `api.github.com/pulumi` return an error directly, and otherwise send a proper response.